### PR TITLE
Add tenantId to azure oauth2 native fs auth

### PR DESCRIPTION
**Pull request by wendigo**
https://trinodb.slack.com/archives/CP1MUNEUX/p1708280746885969

### CODE REVIEW

**Comment by wendigo** [comment]
```
Learn more:
  https://github.com/google/guice/wiki/ERROR_INJECTING_CONSTRUCTOR
Caused by: IllegalArgumentException: Must provide non-null values for tenantId properties in ClientSecretCredentialBuilder
    at ValidationUtil.validate(ValidationUtil.java:50)
    at ClientSecretCredentialBuilder.build(ClientSecretCredentialBuilder.java:129)
    at AzureAuthOauth.<init>(AzureAuthOauth.java:39)
    at AzureAuthOauth.<init>(AzureAuthOauth.java:30)
    at AzureAuthOauth$$FastClassByGuice$$308b03b.GUICE$TRAMPOLINE(<generated>)
    at AzureAuthOauth$$FastClassByGuice$$308b03b.apply(<generated>)
    at DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:82)
    at ConstructorInjector.provision(ConstructorInjector.java:114)
    at ConstructorInjector.access$000(ConstructorInjector.java:33)
    at ConstructorInjector$1.call(ConstructorInjector.java:98)
    at ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:109)
    at LifeCycleModule.provision(LifeCycleModule.java:53)
```

**Comment by wendigo** [comment]
We never tested this in the CI :)

**Comment by findepi** [comment]
> We never tested this in the CI :)

Let's fix that

File: lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureAuthOAuthConfig.java [hunk 1]
```diff
@@ -64,4 +65,17 @@ public AzureAuthOAuthConfig setClientSecret(String clientSecret)
...
         return this;
     }
+
+    @NotEmpty
+    public String getTenantId()
+    {
+        return tenantId;
+    }
+
+    @Config("azure.oauth.tenant-id")
```
Comment (findinpath) [inline lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureAuthOAuthConfig.java:75]: Please add `@ConfigDescription` here

Comment (findepi) [reply lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureAuthOAuthConfig.java:75]: let's do as a follow-up

**Pull request merged**